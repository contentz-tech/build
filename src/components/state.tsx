import { createContext, useContext } from "react";
import { jsx } from "@emotion/core";
import { IState } from "../state";

const StateContext = createContext<IState | null>(null);

interface StateProviderProps {
  state: IState;
  children: JSX.Element[] | JSX.Element;
}

function StateProvider({ state, children }: StateProviderProps) {
  return (
    <StateContext.Provider value={state}>{children}</StateContext.Provider>
  );
}

function useState(): IState {
  const state = useContext(StateContext);
  if (state === null)
    throw new Error(
      "Wrap your component in `StateProvider` and pass a valid state."
    );
  return state;
}

export { StateProvider, useState };
