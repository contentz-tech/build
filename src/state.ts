import { IConfig, config as getConfig } from "./getters/config";
import { articles, IArticles } from "./getters/articles";
import { pages, IPages } from "./getters/pages";
import { slides, ISlides } from "./getters/slides";
import { links, ILink } from "./getters/links";
import { statics } from "./getters/statics";
import { i18n } from "./getters/i18n";

interface IState {
  config: IConfig;
  articles: IArticles;
  pages: IPages;
  slides: ISlides;
  links: ILink[];
  statics: string[];
  i18n: any;
}

async function getState(): Promise<IState> {
  const config: IConfig = await getConfig();
  return {
    config,
    articles: await articles(config),
    pages: await pages(config),
    slides: await slides(config),
    links: await links(),
    statics: await statics(),
    i18n: await i18n(config)
  };
}

export { getState, IState };
