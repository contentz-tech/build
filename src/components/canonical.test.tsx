/** * @jest-environment jsdom */
import "@testing-library/jest-dom/extend-expect";
import { jsx } from "@emotion/core";
import { render } from "@testing-library/react";
import CanonicalURL from "./canonical";
import { StateProvider } from "./state";

const state = {
  config: { language: "en" },
  articles: {
    order: [],
    byPath: {}
  },
  slides: {
    order: [],
    byPath: {}
  },
  statics: [],
  pages: {
    order: [],
    byPath: {}
  },
  links: [],
  i18n: {
    canonicalUrl: { copy: "Originally published at " }
  }
};

describe("CanonicalURL", () => {
  test("Without value", () => {
    const { asFragment } = render(
      <StateProvider state={state}>
        <CanonicalURL />
      </StateProvider>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test("With value", () => {
    const { asFragment, getByText, getByTitle } = render(
      <StateProvider state={state}>
        <CanonicalURL value="https://sergiodxa.com/contentz" />
      </StateProvider>
    );

    expect(asFragment()).toMatchSnapshot();

    expect(getByText(/sergiodxa.com/)).toBeInTheDocument();

    expect(getByTitle(/sergiodxa.com/)).toBeInTheDocument();

    expect(getByTitle(/sergiodxa.com/)).toHaveAttribute(
      "href",
      "https://sergiodxa.com/contentz"
    );
    expect(getByTitle(/sergiodxa.com/)).toHaveAttribute("target", "_blank");
    expect(getByTitle(/sergiodxa.com/)).toHaveAttribute("rel", "canonical");
  });
});
