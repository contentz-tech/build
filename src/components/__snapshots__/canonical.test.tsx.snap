// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CanonicalURL With value 1`] = `
<DocumentFragment>
  .emotion-1 {
  box-shadow: rgba(0,0,0,0.12) 0px 2px 5px 0px;
  background-color: white;
  border-radius: 5px;
  -webkit-transition: all 0.2s ease 0s;
  transition: all 0.2s ease 0s;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  margin: 1.5em -2em 3em;
  padding: 1em 2em;
}

@media (max-width:720px) {
  .emotion-1 {
    box-shadow: none;
    border-radius: 0;
    margin: 0;
    padding: 1em 0.5em;
  }
}

.emotion-0 {
  color: #004BCC;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<div
    class="emotion-1"
  >
    Originally published at 
    <a
      class="emotion-0"
      href="https://sergiodxa.com/contentz"
      rel="canonical"
      target="_blank"
      title="sergiodxa.com"
    >
      <strong>
        sergiodxa.com
      </strong>
    </a>
  </div>
</DocumentFragment>
`;

exports[`CanonicalURL Without value 1`] = `<DocumentFragment />`;
